background/usw-api.js

```diff
@@ -1,29 +1,98 @@
+/* global API msg */// msg.js
 /* global URLS */ // toolbox.js
-
+/* global tokenMan */
 'use strict';
 
-/* exported retrieveStyleInformation */
-async function retrieveStyleInformation(token) {
-  return (await (await fetch(`${URLS.usw}api/style`, {
-    method: 'GET',
-    headers: new Headers({
-      'Authorization': `Bearer ${token}`,
-    }),
-    credentials: 'omit',
-  })).json()).data;
-}
+const uswApi = (() => {
+
+  //#region Internals
+
+  class TokenHooks {
+    constructor(id) {
+      this.id = id;
+    }
+    keyName(name) {
+      return `${name}/${this.id}`;
+    }
+    query(query) {
+      return Object.assign(query, {vendor_data: this.id});
+    }
+  }
+
+  async function linkStyle(style, sourceCode) {
+    const {id} = style;
+    const metadata = await API.worker.parseUsercssMeta(sourceCode).catch(console.warn) || {};
+    const uswData = Object.assign({}, style, {metadata, sourceCode});
+    API.data.set('usw' + id, uswData);
+    const token = await tokenMan.getToken('userstylesworld', true, new TokenHooks(id));
+    const info = await uswFetch('style', token);
+    const data = style._usw = Object.assign({token}, info);
+    await uswSave(style);
+    return data;
+  }
+
+  async function uswFetch(path, token, opts) {
+    opts = Object.assign({credentials: 'omit'}, opts);
+    opts.headers = Object.assign({Authorization: `Bearer ${token}`}, opts.headers);
+    return (await (await fetch(`${URLS.usw}api/${path}`, opts)).json()).data;
+  }
+
+  /** Uses a custom method when broadcasting and avoids needlessly sending the entire style */
+  async function uswSave(style) {
+    const {id, _usw} = style;
+    await API.styles.save(style, {broadcast: false});
+    msg.broadcastExtension({method: 'uswData', style: {id, _usw}});
+  }
+
+  //#endregion
+  //#region Exports
+
+  return {
+    /**
+     * @param {number} id
+     * @param {string} sourceCode
+     * @return {Promise<string>}
+     */
+    async publish(id, sourceCode) {
+      const style = await API.styles.get(id);
+      const data = (style._usw || {}).token
+        ? style._usw
+        : await linkStyle(style, sourceCode);
+      return uswFetch(`style/${id}`, data.token, {
```

Comment:
```suggestion
      return uswFetch(`style/${data.id}`, data.token, {
```

You were passing's the internal ID that stylus's uses. Which is 99% not the corresponding ID to the ID on USw. And most-likely causing that it doesn't belong to you error.

Reply:
Yay, thanks, it works now! Indeed, I incorrectly inlined the previiously separate function uploadStyle.